### PR TITLE
[SYCL][LIT] Add checks if half/double type supported in reduction LIT…

### PR DESCRIPTION
**Pull request by v-klochkov**
… tests

Signed-off-by: Vyacheslav N Klochkov <vyacheslav.n.klochkov@intel.com>

### CODE REVIEW

File: sycl/test/reduction/reduction_nd_s0_dw.cpp [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // UNSUPPORTED: cuda
 // Reductions use work-group builtins not yet supported by CUDA.
 //
-// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -fsycl-device-code-split=per_kernel -o %t.out
```
Comment (bjoernknafla) [inline sycl/test/reduction/reduction_nd_s0_dw.cpp:4]: Is the `-fsycl-device-code-split=per_kernel` required as it is not supported by CUDA (on top of the work-group built-in) and might make enabling the test in the future harder as both features are required?

File: sycl/test/reduction/reduction_nd_s0_dw.cpp [hunk 2]
```diff
@@ -21,6 +21,10 @@ class SomeClass;
 
 template <typename T, int Dim, class BinaryOperation>
 void test(T Identity, size_t WGSize, size_t NWItems) {
+  queue Q;
+  if (!isSupportedType<T>(Q.get_device()))
```
Comment (bjoernknafla) [inline sycl/test/reduction/reduction_nd_s0_dw.cpp:25]: Not for this MR though @bader  brought up restructuring the test system in the future and this is an example of us probably wanting feature detection to not even need to run testst that require specific features/extensions.

File: sycl/test/reduction/reduction_nd_s0_dw.cpp (hunk 1 above)
Comment (v-klochkov) [reply sycl/test/reduction/reduction_nd_s0_dw.cpp:4]: I discovered that 'has_extension("cl_khr_double")' helps to not run the code with 'double' type if device does not support 'double', but the test still fails without -fsycl-device-code-split=per_kernel.

Ok, the alternatives to this PR is a) not testing 'double' and 'half' at all, or b) move them to separate test(s).

**Comment by v-klochkov** [comment]
I uploaded an alternative fix/PR here: https://github.com/intel/llvm/pull/1912

**Pull request closed without merging**